devito/ir/iet/visitors.py

```diff
@@ -315,6 +317,11 @@ def visit_Operator(self, o):
                                                     self._args_decl(i.root.parameters)))
                 efuncs.extend([i.root.ccode, blankline])
 
+        for i in o._callables:
```

Comment:
no need for ``_callable``; just use ``_func_table``. Sorry, I should have told you about this

Reply:
I tried using `_func_table` but it was doing some processing on it that caused an exception

Reply:
such as ...? definitely we don't want two different objects to store callables. 

a value in the ``_func_table`` is an object of type MataCall, such as the one used here: https://github.com/opesci/devito/blob/master/devito/operator.py#L281

left argument is the ``root`` of an iet (eg/typically a Callable object)
right argument is True if the IET of the callable is "local" ie produced by devito and therefore available as an IET. If false, it's like a call to a third-party library

Reply:
https://github.com/opesci/devito/blob/master/devito/operator.py#L296 it fails on this line. So I assume I should mark it as non-local (I was setting it as `True` since it is actually generated by devito) so that this step is not executed?

Reply:
ah yes, perhaps non-local will do the trick

Reply:
how does it fail on that line? what's the error message? we need to execute that line otherwise if we have for example scalar temporaries they won't get declared (ie, `a = 0`  -> `float a = 0`)